locals.tf

```diff
@@ -7,6 +7,21 @@
 
 locals {
   kubeconfig_filename = "work/kubeconfig.yaml"
-  map_roles           = join("", data.template_file.map_roles[*].rendered)
+  map_roles = templatefile("${path.module}/templates/config_map_aws_map_roles.yaml.tpl",
+  {
+    map_roles = var.map_roles
+  })
+  kubeconfig = templatefile("${path.module}/templates/kubeconfig.tpl",
+  {
+    server                     = join("", aws_eks_cluster.this[*].endpoint)
```

Comment:
I guess there will always only be one aws_eks_cluster, so maybe you can skip `join()` and just reference `[0]`? If there is more than one the join would result in a useless kubeconfig anyway.

Reply:
Good point. I've updated the PR to just use `aws_eks_cluster.this.endpoint`